### PR TITLE
Fixed issue #775

### PR DESCRIPTION
**Pull request by wongjiahau**
Refer issue #775

### CODE REVIEW

**Comment by CameronAavik** [comment]
I think the reason that the original code handled 24 that way was because it was being used by the GenerateButtons() method (the one with no arguments) which will pass in 24 when Is24Hours is true. In this circumstance, we do want 24 to be represented as 0, not 24.

One solution to fix this would be to say that i = i % numItems. But this breaks when we use 12 hour time, which is when we want 1, 2, 3, ..., 12 to be shown and not 0.

So the actual solution instead should be passing down the correct values instead of fixing them inside this method.

**Comment by wongjiahau** [comment]
@CameronAavik Thanks for the reminder, I just realized that.

File: MaterialDesignThemes.Wpf/Clock.cs [hunk 1]
```diff
@@ -317,8 +317,13 @@ private void HourReadOutPartNameOnPreviewMouseLeftButtonDown(object sender, Mous
 				button.SetBinding(Canvas.LeftProperty, GetBinding("X", button));
 				button.SetBinding(Canvas.TopProperty, GetBinding("Y", button));
 
-				button.Content = (i == 60 ? 0 : (i == 24 ? 0 : i)).ToString(format);
-				canvas.Children.Add(button);
+			    if (Is24Hours) {
+			        button.Content = (i == 60 ? 0 : (i == 24 ? 0 : i)).ToString(format);
+			    }
+			    else {
```
Comment (mokeev1995) [inline MaterialDesignThemes.Wpf/Clock.cs]: You've mixed identation. Switch to spaces.

Comment (wongjiahau) [inline MaterialDesignThemes.Wpf/Clock.cs]: @mokeev1995 Er... Do you mind to tell me how to fix it? Because I'm not sure how to fix that

**Comment by Keboo** [comment]
This was fixed by #1155

**Pull request closed without merging**